snapshot/windows/windows.go

```diff
@@ -40,45 +79,304 @@ func NewSnapshotter(root string) (snapshot.Snapshotter, error) {
+	ctx, t, err := o.ms.TransactionContext(ctx, true)
+	if err != nil {
+		return snapshot.Info{}, err
+	}
+
+	info, err = storage.UpdateInfo(ctx, info, fieldpaths...)
+	if err != nil {
+		t.Rollback()
+		return snapshot.Info{}, err
+	}
+
+	if err := t.Commit(); err != nil {
+		return snapshot.Info{}, err
+	}
+
+	return info, nil
 }
 
-func (o *Snapshotter) Usage(ctx context.Context, key string) (snapshot.Usage, error) {
-	panic("not implemented")
+func (o *snapshotter) Usage(ctx context.Context, key string) (snapshot.Usage, error) {
+	ctx, t, err := o.ms.TransactionContext(ctx, false)
+	if err != nil {
+		return snapshot.Usage{}, err
+	}
+	defer t.Rollback()
+
+	_, info, usage, err := storage.GetInfo(ctx, key)
+	if err != nil {
+		return snapshot.Usage{}, err
+	}
+
+	if info.Kind == snapshot.KindActive {
+		//du, err := fs.DiskUsage(o.getSnapshotDir(id))
+		//if err != nil {
+		//	return snapshot.Usage{}, err
+		//}
+		du := fs.Usage{
+			Size: 0,
+		}
+		usage = snapshot.Usage(du)
+	}
+
+	return usage, nil
 }
 
-func (o *Snapshotter) Prepare(ctx context.Context, key, parent string, opts ...snapshot.Opt) ([]mount.Mount, error) {
-	panic("not implemented")
+func (o *snapshotter) Prepare(ctx context.Context, key, parent string, opts ...snapshot.Opt) ([]mount.Mount, error) {
+	return o.createSnapshot(ctx, snapshot.KindActive, key, parent, opts)
 }
 
-func (o *Snapshotter) View(ctx context.Context, key, parent string, opts ...snapshot.Opt) ([]mount.Mount, error) {
-	panic("not implemented")
+func (o *snapshotter) View(ctx context.Context, key, parent string, opts ...snapshot.Opt) ([]mount.Mount, error) {
+	return o.createSnapshot(ctx, snapshot.KindView, key, parent, opts)
 }
 
 // Mounts returns the mounts for the transaction identified by key. Can be
 // called on an read-write or readonly transaction.
 //
 // This can be used to recover mounts after calling View or Prepare.
-func (o *Snapshotter) Mounts(ctx context.Context, key string) ([]mount.Mount, error) {
-	panic("not implemented")
+func (o *snapshotter) Mounts(ctx context.Context, key string) ([]mount.Mount, error) {
+	ctx, t, err := o.ms.TransactionContext(ctx, false)
+	if err != nil {
+		return nil, err
+	}
+	s, err := storage.GetSnapshot(ctx, key)
+	t.Rollback()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get snapshot mount")
+	}
+	return o.mounts(s), nil
 }
 
-func (o *Snapshotter) Commit(ctx context.Context, name, key string, opts ...snapshot.Opt) error {
-	panic("not implemented")
+func (o *snapshotter) Commit(ctx context.Context, name, key string, opts ...snapshot.Opt) error {
+	ctx, t, err := o.ms.TransactionContext(ctx, true)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err != nil {
+			if rerr := t.Rollback(); rerr != nil {
+				log.G(ctx).WithError(rerr).Warn("Failure rolling back transaction")
+			}
+		}
+	}()
+
+	// grab the existing id
```

Comment:
todo missing?

Reply:
ID is only needed to get the disk usage. Removed.